contrib/p4/docker_files/ptf_tester/tests/test_no_fail.py

```diff
@@ -0,0 +1,172 @@
+# SPDX-FileCopyrightText: 2020-2021 CASTOR Software Research Centre
+# <https://www.castor.kth.se/>
+# SPDX-FileCopyrightText: 2020-2021 Johan Paulsson
+
+# SPDX-License-Identifier: Apache-2.0
+
+import ptf
+from ptf.base_tests import BaseTest
+from ptf import config
+import ptf.testutils as testutils
+from scapy.all import *
+import re
+
+TYPE_MYTUNNEL = 0x1212
+TYPE_IPV4 = 0x0800
+
+
+class MyTunnel(Packet):
+    name = "MyTunnel"
+    fields_desc = [ShortField("pid", 0), ShortField("dst_id", 0)]
+
+    def mysummary(self):
+        return self.sprintf("pid=%pid%, dst_id=%dst_id%")
+
+
+class DataplaneBaseTest(BaseTest):
+    def __init__(self):
+        BaseTest.__init__(self)
+
+    def setUp(self):
+        self.dataplane = ptf.dataplane_instance
+        self.dataplane.flush()
+        if config["log_dir"] != None:
+            filename = os.path.join(config["log_dir"], str(self)) + ".pcap"
+            self.dataplane.start_pcap(filename)
+
+    def tearDown(self):
+        if config["log_dir"] != None:
+            self.dataplane.stop_pcap()
+
+
+class IPV4OneSwitchTest(DataplaneBaseTest):
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def runTest(self):
+        # log_to_file("Staring first test")
+        pkt = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.2",
+            ip_ttl=64,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+        pkt_expected = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.2",
+            ip_ttl=63,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+        try:
+            testutils.send_packet(self, (0, 0), pkt)
+        except Exception as e:
```

Comment:
Are you sure that you want to capture and swallow all exceptions here? Shouldn't they propagate and make the test fail?